main.py

```diff
@@ -548,97 +598,101 @@ def generate_tracks(cars_to_use, board_to_use, ints_to_use, available_tracks, he
             continue
         if amt_placed_decoy == -1:
             return
-        generate_tracks(list(car_combos[combo_num]), board_to_pass, ints_to_pass,
-                        available_tracks - amt_placed_decoy, np.array(heatmaps), [list(solved[0]), list(solved[1])],
-                        stalled_to_pass, np.array(switch_queue), list(station_stalled),
-                        np.array(crashed_decoys), mvmts_since_solved, available_semaphores - placed_semaphores,
-                        heatmap_limits_pass)
+        yield (
+                list(car_combos[combo_num]), board_to_pass, ints_to_pass,
+                available_tracks - amt_placed_decoy, np.array(heatmaps), [list(solved[0]), list(solved[1])],
+                stalled_to_pass, np.array(switch_queue), list(station_stalled),
+                np.array(crashed_decoys), mvmts_since_solved, available_semaphores - placed_semaphores,
+                heatmap_limits_pass
+        )
 
 
 # use 11-8b for visualizer
 # 10-7 problem needs to be fixed where semaphores only check for last generated car
-for lvl in [lc.levels["10-7"]]:
-# for key in lc.world7:
-#     lvl = lc.world7[key]
-#     print(key)
-    lowestTracksRemaining = -1
-    semaphoresRemaining = -1
-    # check how many parameters are in the level before doing it to load it properly
-    # I will change this eventually because it's more consistent that way, and I'll have to redo every single
-    # level setup, but this works for now.
-    if len(lvl) == 4:
-        board, cars, maxTracks, interactions = lvl
-        decoys = []
-        semaphores = 0
-        ncars = []
-    elif len(lvl) == 5:
-        board, cars, maxTracks, interactions, decoys = lvl
-        if decoys is None:
-            decoys = []
-        semaphores = 0
-        ncars = []
-    elif len(lvl) == 6:
-        board, cars, maxTracks, interactions, decoys, semaphores = lvl
-        if decoys is None:
+for level_name in ["10-7"]:
+    with PrinterProfile(name=level_name):
+        lvl = lc.levels[level_name]
+    # for key in lc.world7:
```

Comment:
Everything from here is just whitespace changes because everyhting is indented one level for the profiler context